mino/minows/mod.go

```diff
@@ -87,18 +97,74 @@ func (m *minows) CreateRPC(name string, h mino.Handler, f serde.Factory) (mino.R
 	if !pattern.MatchString(name) {
 		return nil, xerrors.Errorf("invalid name: %s", name)
 	}
-	if _, ok := m.rpcs[name]; ok {
+	if _, found := m.rpcs[name]; found {
 		return nil, xerrors.Errorf("already exists rpc: %s", name)
 	}
-	uri := strings.Join(append(m.segments, name), "/")
-	// TODO wrap mino.Handler in network.StreamHandler
-	//  to handle Call() with Process() and Stream() with Stream()
-	// TODO m.host.SetStreamHandler(identity, h)
-	// TODO when to return pointer vs struct?
-	return rpc{
-		uri:     protocol.ID(uri),
+	// create rpc
+	uri := strings.Join(append(m.namespace, name), "/")
+	r := &rpc{
+		logger:  m.logger.With().Str("rpc", uri).Logger(),
+		uri:     uri,
 		mino:    m,
 		factory: f,
 		context: m.context,
-	}, nil
+	}
+	// TODO need to be thread safe?
+	m.rpcs[name] = nil
+	// start listening
+	m.host.SetStreamHandler(protocol.ID(uri+PostfixCall),
+		r.createCallHandler(h))
+	m.host.SetStreamHandler(protocol.ID(uri+PostfixStream),
+		r.createStreamHandler(h))
+	// TODO when to return pointer vs struct?
+	return r, nil
```

Comment:
https://bryanftan.medium.com/accept-interfaces-return-structs-in-go-d4cab29a301b

But I'm not sure how dela handles this in other modules.

Reply:
returns pointer now